src/GitHub/Lens.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE PolyKinds     #-}
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+{- |
+Copyright: (c) 2021 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+Utilities to provide general @lens@-like API for creating GraphQL
+queries.
+-}
+
+module GitHub.Lens
+    ( HasOwnerName (..)
+    , HasLimit (..)
+    , HasStates (..)
+
+      -- * Parameter types
+    , ArgsType (..)
+
+      -- * Internas
+    , Delete
+    ) where
+
+import Data.Kind (Type)
+import Data.List.NonEmpty (NonEmpty)
+import Data.Text (Text)
+import Prolens (Lens)
+
+import GitHub.GraphQL (State)
+
+
+{- | Type of fields for arguments.
+-}
+data ArgsType
+    = ArgsOwner
+    | ArgsName
+    | ArgsLimit
+    | ArgsStates
```

Comment:
Maybe this data type should belong to a separate module and be documented a bit better, clarifying the purpose, relationship to `*Args` like data types? Or maybe later we can add documentation about how to add the support of a new API call?

Reply:
A better name would be `RequiredField` 😄